### DISCUSSION TITLE
Different behavior of fitz.Pixmap() between `PyMuPDF` 1.23.8 and 1.23.15

### DISCUSSION THREAD

**Discussion by dothinking** [pymupdf/PyMuPDF#3057] [Looking for help] [answered]
With upgrading `PyMuPDF` from 1.23.8 to 1.23.15, fitz.Pixmap() doesn't work as expected. Do I miss something, or if it's an upstream issue.

[sample.pdf](https://github.com/pymupdf/PyMuPDF/files/13974803/sample.pdf)

## PyMuPdf 1.23.8

```python
>>> import fitz
>>> fitz.version
('1.23.8', '1.23.7', '20231219000001')
>>> doc = fitz.Document('sample.pdf')
>>> doc[0].get_images(full=True)
[(17, 0, 1731, 1331, 8, 'DeviceCMYK', '', 'Im0', 'DCTDecode', 0)]
>>> pix = fitz.Pixmap(doc, 17)
>>> pix.colorspace
Colorspace(CS_CMYK) - DeviceCMYK
>>> pix = fitz.Pixmap(fitz.csRGB, pix)
>>> pix.save('sample-1.png')                      <- perfect
```

## PyMuPdf 1.23.15

```python
>>> import fitz
>>> fitz.version
('1.23.15', '1.23.9', '20240116000001')
>>> doc = fitz.Document('sample.pdf')
>>> doc[0].get_images(full=True)
[(17, 0, 1731, 1331, 8, 'DeviceCMYK', '', 'Im0', 'DCTDecode', 0)]    <- color space = DeviceCMYK
>>> pix = fitz.Pixmap(doc, 17)
>>> pix.colorspace
Colorspace(CS_RGB) - DeviceRGB                     <- color space seems changed to DeviceRGB
>>> pix = fitz.Pixmap(fitz.csRGB, pix)
>>> pix.save('sample-2.png')                   <- the png color is different from the source
```

**Comment by JorjMcKie** [comment]
Thanks for posting this. Looking now.
Background:
The difference between these PyMuPDF versions is that since 1.23.9 we are an "rebased" architecture. At 1.23.8, `import fitz` pointed to the "classic" implementation.
Now, `import fitz` points to the "rebased" implementation and the classic one is available via `import fitz_old as fitz`.
Let me check / confirm your observation. We will possibly make an issue from your post.

**Comment by JorjMcKie** [comment, marked as answer]
@dothinking - confirming your observation:
The Pixmap made from the image is indeed converted to RGB but shouldn't: it should be CMYK.
Converting to a bug.

**Reply by dothinking** [reply]
Thanks for your quick reply. Then I should downgrade for now and waiting for a fix.

**Comment by julian-smith-artifex-com** [comment]
This is fixed in release 1.23.16.
